app/src/androidTest/java/com/example/android/justjava/Helpers/Matchers/ToastMatcher.kt

```diff
@@ -0,0 +1,27 @@
+package com.example.android.justjava.Helpers.Matchers
+
+import android.os.IBinder
+import android.support.test.espresso.Root
+import android.view.WindowManager
+
+import org.hamcrest.Description
+import org.hamcrest.TypeSafeMatcher
+
+class ToastMatcher : TypeSafeMatcher<Root>() {
+
+    override fun describeTo(description: Description) {
+        description.appendText("is toast")
+    }
+
+    public override fun matchesSafely(root: Root): Boolean {
+        val type = root.windowLayoutParams.get().type
+        if (type == WindowManager.LayoutParams.TYPE_TOAST) {
```

Comment:
Not sure if you noticed but TYPE_TOAST always has a line through it and says that it's deprecated. I don't remember what they suggested to replace it with but I was never able to get it to work with the replacement... So I just left it as TYPE_TOAST because it works even if they say it is deprecated. Never got a chance to research that much.